homeassistant/components/recorder/migration.py

```diff
@@ -1089,6 +1108,47 @@ def _apply_update(  # noqa: C901
             "states",
             [f"last_reported_ts {_column_types.timestamp_type}"],
         )
+    elif new_version == 44:
+        columns: tuple[str, ...]
+        identity_sql = (
+            "NOT NULL AUTO_INCREMENT"
+            if engine.dialect.name == SupportedDialect.MYSQL
+            else ""
+        )
+        # First drop foreign key constraints
+        foreign_columns = (
+            ("events", ("data_id", "event_type_id")),
+            ("states", ("event_id", "old_state_id", "attributes_id", "metadata_id")),
+            ("statistics", ("metadata_id",)),
+            ("statistics_short_term", ("metadata_id",)),
+        )
+        for table, columns in foreign_columns:
+            for column in columns:
+                _drop_foreign_key_constraints(session_maker, engine, table, [column])
+
+        # Then modify the ID columns
+        id_columns = (
+            ("events", ("event_id",)),
+            ("event_data", ("data_id",)),
+            ("event_types", ("event_type_id",)),
+            ("states", ("state_id",)),
+            ("state_attributes", ("attributes_id",)),
+            ("states_meta", ("metadata_id",)),
+            ("statistics", ("id",)),
+            ("statistics_short_term", ("id",)),
+            ("statistics_meta", ("id",)),
+            ("recorder_runs", ("run_id",)),
+            ("schema_changes", ("change_id",)),
+            ("statistics_runs", ("run_id",)),
+        )
```

Comment:
Its a bit unexpected that the columns are a tuple as there should only in theory be one per table?